### ISSUE TITLE
More Stuff: Add support for Subcommands, similar to GL::SubCommand

### ISSUE THREAD

**Issue by tabulon** [perlancar/perl-Getopt-Long-More#27]
## SYNOPSIS

Add support for `subcommands`.

Here's a conventional description of  `subcommands` taken from the [CLI11][CLI11/subcommands] project: 

>Subcommands are keyword that invoke a new set of options and features. For example, the `git` command has a long series of subcommands, like `add` and `commit`. Each can have its own options and implementations.

## The WHY

Subcommands (sometimes simply called 'commands') have been quite popular for some time now. 

While certain aspects of subcommands (such as the actual dispatch) would arguably be best treated by a **CLI framework**, certain other aspects  (such as the need to specify and parse command line options) clearly have an interplay with 'GetOpt' functionality.

If the options parser (GLM, in this case) does not provide simple mechanisms for supporting this sort of thing, then many wheels need to be re-invented by CLI frameworks, or worse, command scripts themselves.

## The WHAT

GLM itself should not be providing a full-fledged `dispatch mechanism`, which is probably the job of a higher level **CLI framework**. 

Instead, GLM should simply lay out the groundwork for easily building such a mecanism.

## The API MECHANISM

**CONSIDER**: Adopting a `subcommand` API that is conceptually quite similar to that of ([Getopt::Long::SubCommand][Getopt::Long::SubCommand]),.

Meanwhile, GLM's `GetOptions()` needs to remain compatible with GOL, which provides additional incentive to investigate an approach à la `cmdpec()` as mentioned in #18.

---

[Getopt::Long::Descriptive]: https://metacpan.org/pod/Getopt::Long::Descriptive
[Getopt::Long::SubCommand]: https://metacpan.org/pod/Getopt::Long::SubCommand

[CLI11]: https://cliutils.gitlab.io/
[CLI11/subcommands]: https://cliutils.gitlab.io/CLI11Tutorial/chapters/subcommands.html

**Comment by perlancar** [comment]
I agree that cmdspec() (#18) is the mechanism to introduce this feature into GLM in a GoL-compatible way.

**Comment by tabulon** [comment]
Which one of the below interfaces do you like better?

a) 

```perl
GetOptions( cmdspec(
   ...
   subcommands => { ... },   # or simply, "commands"
   ...
));
```

b) 

```perl
GetOptions( cmdspec(
  ...
  options  => [
    'foo' => optspec (...),
    '&list'  => cmdspec (    # not sure about the &
        ...
     ),
  ],
  ...
));
```
I see merits for both, but `(a)` is probably the way to go. It's just simpler.

**Comment by perlancar** [comment]
I prefer a) and "subcommands". Command, IMO, should refer to the program/script, to avoid confusion.

**Comment by tabulon** [comment]
Agreed.
